src/utils/auth.js

```diff
@@ -1,58 +1,61 @@
-export const BASE_URL = 'http://localhost:3000';
+import { BASE_URL } from "./constants";
 
-function getResponseData(response, errorText) {
-  if (response.ok) {
-    return response.json();
+function getResponseData(res) {
+  if (res.ok) {
+    return res.json();
   } else {
-    throw new Error(errorText);
+    return Promise.reject(`Error in response from the server: ${res.status}`)
   }
 }
-export const register = (email, password, name) => {
+//signup
+export function register(email, password, name) {
   return fetch(`${BASE_URL}/signup`, {
     method: 'POST',
     headers: {
+      Accept: "application/json",
       "Content-Type": "application/json",
     },
     body: JSON.stringify({ email, password, name }),
   })
-    .then((response) =>
-      getResponseData(response, 'Unsuccessful register')
+    .then((res) => {
+      return getResponseData(res)}
     ).catch((err) => {
       console.log(err);
     })
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very end of any request and this end is in  (`App.js`)  after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.

Reply:
Please, delete `catch` from line 22 in `auth.js`


* You haven't corrected something https://github.com/MJourno/news-explorer-frontend/pull/3#discussion_r1296727063